### DISCUSSION TITLE
Multi-Tenancy: Nested Relationships

### DISCUSSION THREAD

**Discussion by jpau-besmartee** [filamentphp/filament#7569] [Help]
Really loving the new tenant functionality in V3 and have been exploring it to see what all it can handle.

I'm wondering though, does the tenant functionality have a way of handling records that are nested more than 1 layer deep in an application?

An example of what I mean is, say we have a **Team** model which is our tenant model. **Teams** can have many **Authors**, **Authors** can have many **Posts**, and **Posts** can have many **Comments**.

What would the approach on how to handle **Comments** when a comment is associated to a **Team** _through_ **Posts** _through_ **Authors**?

From what I understand in the documentation, using a [Relation Manager](https://filamentphp.com/docs/3.x/panels/resources/relation-managers) is the approach when dealing with a single nest (**Authors** to **Posts**). However, creating a **PostResource** that has a relation manager for the **Comments** yields the tenant functionality trying to create the **Post** for the **Team** when I really want to be able to select an _author_ for the **Post**.

Would love to hear your suggestions!

**Comment by flashmediasolutions** [comment]
Digging around the same issue. Works great 1 deep resource with relation managers, but unable to breakout second level items on their own.

Using your example, I can't figure out how to show all Posts as a resource.

**Comment by juliangums** [comment]
I managed to do this by defining BelongsToThrough relationships https://github.com/staudenmeir/belongs-to-through
I did it even 4 layers deep and it works just fine.
